bofire/benchmarks/benchmark.py

```diff
@@ -74,6 +76,28 @@ def _single_run(
     initial_sampler: Optional[Callable[[Domain], pd.DataFrame]],
     n_candidates_per_proposals: int,
 ) -> Tuple[Benchmark, pd.Series]:
+    def autosafe_results(benchmark):
+        """Safes results into a .json file to prevent data loss during time-expensive optimization runs.
+        Autosave should operate every 10 iterations.
+
+        Args:
+            benchmark: Benchmark function that is suposed be evaluated.
+        """
+
+        benchmark_name = benchmark.__class__.__name__
+        # Create a folder for autosaves, if not already exists.
+        if not os.path.exists("bofire_autosaves/" + benchmark_name):
+            os.makedirs("bofire_autosaves/" + benchmark_name)
+
+        filename = (
+            "bofire_autosaves/" + benchmark_name + "/run" + str(run_idx) + ".json"
+        )
+        # TODO: Save domain as a whole into a .json. Possible when domain can be parsed into a dict (waiting for an update)
+        # parsed_domain = benchmark.domain.dict()
+        parsed_domain = benchmark.domain.experiments.to_json(orient="split")
```

Comment:
Benni promised me Thursday for this method. As soon as it is there we merge it in here.